pkg/dogstatsd/listeners/named_pipe_windows.go

```diff
@@ -63,22 +62,64 @@ func newNamedPipeListener(
 	listener := &NamedPipeListener{
 		pipe:          pipe,
 		packetManager: packetManager,
-		connections:   make(map[net.Conn]struct{}),
+		connections: &namedPipeConnections{
+			newConn:         make(chan net.Conn),
+			connToClose:     make(chan net.Conn),
+			closeAllConns:   make(chan struct{}),
+			allConnsClosed:  make(chan struct{}),
+			activeConnCount: 0,
+		},
 	}
 
 	log.Debugf("dogstatsd-named-pipes: %s successfully initialized", pipe.Addr())
 	return listener, nil
 }
 
+type namedPipeConnections struct {
+	newConn         chan net.Conn
+	connToClose     chan net.Conn
+	closeAllConns   chan struct{}
+	allConnsClosed  chan struct{}
+	activeConnCount int32
+}
+
+func (l *namedPipeConnections) handleConnection() {
+	connections := make(map[net.Conn]struct{})
+	requestStop := false
+	for stop := false; !stop; {
+		select {
+		case conn := <-l.newConn:
+			connections[conn] = struct{}{}
+			atomic.AddInt32(&l.activeConnCount, 1)
+		case conn := <-l.connToClose:
+			conn.Close()
+			delete(connections, conn)
```

Comment:
Unit test pass because 
```
for listener.getActiveConnectionsCount() > 1 {	
		time.Sleep(100 * time.Millisecond)	
	}
```
should be 
```
for listener.getActiveConnectionsCount() > 0 {	
		time.Sleep(100 * time.Millisecond)	
	}
```